src/infrastructure/healthchecks/healthchecks.ts

```diff
@@ -0,0 +1,157 @@
+import { types } from 'node:util'
+import type { Either } from '@lokalise/node-core'
+import { Gauge } from 'prom-client'
+
+import type { PrismaClient } from '@prisma/client'
+import type Redis from 'ioredis'
+import { FifoMap } from 'toad-cache'
+import type { CommonDependencies } from '../commonDiConfig.js'
+
+export type SupportedHealthchecks = 'redis' | 'postgres'
+
+export const healthcheckResultStore = new FifoMap<boolean>(5, 20)
+export const healthcheckLatencyStore = new FifoMap<number | undefined>(5, 20)
+
+export function getHealthcheckResult(healthcheck: SupportedHealthchecks): boolean {
+  // if we don't yet have results, we consider that to be a success, in order to avoid arbitrary restarts
```

Comment:
as mentioned in https://github.com/lokalise/node-service-template/pull/724/files#r1698146056, I'm slightly concerned about the word "yet" - what if it's actually "anymore" instead?